### PR TITLE
have Gnu.gpg() install gnupg to provide gpg2 command needed

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

**Comment by gartung** [comment]
@mathstuf

**Comment by mathstuf** [comment]
The test shouldn't need to be predicated on gpg2 being installed or not anymore.

**Comment by gartung** [comment]
This fails in the pytest environment anyway.

File: lib/spack/spack/util/gpg.py [hunk 1]
```diff
@@ -31,18 +31,29 @@
 
 GNUPGHOME = spack.gpg_path
 
+def get_gpg2():
+	 dir = os.getcwd()
+	 gnupg_spec = spack.cmd.parse_specs("gnupg", concretize=True)[0]
+	 gnupg = spack.repo.get(gnupg_spec)
+	 if not gnupg.installed:
+		  gnupg.do_install()
+	 os.chdir(dir)
```
Comment (mathstuf) [inline lib/spack/spack/util/gpg.py]: Is there a context manager for the current directory in the standard library?

Comment (gartung) [inline lib/spack/spack/util/gpg.py]: I don't know. @tgamblin might know.

Comment (scheibelp) [inline lib/spack/spack/util/gpg.py]: llnl.util.filesystem.working_dir

File: lib/spack/spack/util/gpg.py [hunk 2]
```diff
@@ -31,15 +31,16 @@
...
-		  gnupg.do_install()
-	 os.chdir(dir)
-	 gnupg_executable = os.path.join(gnupg.prefix, "bin", "gpg2")
-	 return gnupg_executable
+    dir = os.getcwd()
+    gnupg_spec = spack.cmd.parse_specs("gnupg", concretize=True)[0]
+    gnupg = spack.repo.get(gnupg_spec)
+    if not gnupg.installed:
+        gnupg.do_install()
+    os.chdir(dir)
```
Comment (mathstuf) [inline lib/spack/spack/util/gpg.py]: This should use the chdir helper from the `llnl` package here.

Comment (gartung) [inline lib/spack/spack/util/gpg.py]: I am not sure why it is even needed so I will remove the chdir.

Comment (mathstuf) [inline lib/spack/spack/util/gpg.py]: That works just as well too :) .

**Comment by gartung** [comment]
@mathstuf I can't get 'spack test gpg' to pass unless I run
cp spack/var/spack/gpg.mock/keys/external.key spack/var/spack/gpg

**Comment by mathstuf** [comment]
It was broken by f159246d1d073ae7230b5412a4a2a20eac7f49c2. The test used to set internal variables to override the paths used by the commands so it could act in its own sandbox.

@tgamblin Better to revert that and figure out how to allow the test to continue doing its internal redirections with the new command calling feature?

**Comment by mathstuf** [comment]
Or it looks like it's just one command that needs manually parsed (the import one). Maybe just revert that one command call?

**Comment by gartung** [comment]
@mathstuf see #4953

**Pull request closed without merging**